### DISCUSSION TITLE
근원적인 생각

### DISCUSSION THREAD

**Discussion by peterkangdev** [naverz/zepeto-studio-kor#138] [5. Q&A]
월드 개발을 하면서 여러가지 기본적인 어려움에 봉착하면서 문득 드는 근원적인 생각이 있어서 질문드립니다. 
왜. C# 을 버리고, typescript 를 썼을까? 
이런 생각이 문득 들었습니다. 제페토 입장에서도 생각해 보면 여러가지 제약이 필요해서 그런 것이 아닌가...
아무리 생각해봐도 월드 확장에서는 크게 도움이 되지 않을것 같다는 저만의 생각을 해 봅니다. (물론 더 큰 생각이 있으시겠지요)
수 많은 asset store 를 사용하지 못하는 것이 매우 안타깝고, 슬프지만.. 
어쩄든 오늘도 월드 개발하시는 모든 분들... 수고들 많이 하십니다.

**Comment by x4cf** [comment]
다양한 OS와 기기에서 작동하는 프로그램 위에 올리기 용이하게 하기위해 타입스크립트를 선택한것은 아닐까해요, 대부분의 메타버스 에디터가 언어를 하나 거치더라구요

**Comment by cheolhui-lee** [comment]
안녕하세요.
저희도 C#을 사용하실 수 있게 제공해 드릴 수 있었다면 더욱 많은 유니티 개발자들과 제페토 월드의 생태계를 함께 할 수 있다고 생각합니다.
다만 바이너리로 빌드를 해야 하는 C# 언어의 특성상 C#으로 제작된 월드를 앱 내부에 포함하지 않고 동적으로 로딩하여 구현하는 것이 쉽지가 않습니다.

그래서 불가피하게 런타임에서 해석과 실행이 가능하도록 스크립트 엔진을 탑재하여,
앱 업데이트 배포와 무관하게 필요할 때 월드를 앱에서 직접 다운로드하고 실행할 수 있는 JavaScript를 선택하게 되었습니다.

하지만 막상 JavaScript를 선택하고 개발을 해보니, C#과 문법이 너무 달라서 개발에 어려움이 매우 컸는데요.
그 문제를 해결하고자 그나마 C#과 비슷한 부분이 많은 TypeScript를 도입하게 되었습니다.
TypeScript는 JavaScript로 변환되어 런타임에서 구동이 가능합니다.

그럼 불가피한 선택이었던 점 이해 부탁드립니다.
